backend/app/services/metrics_collector.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import time
+from collections import deque
+from datetime import UTC, datetime
+from threading import Lock
+from typing import NamedTuple
+from uuid import UUID
+
+from app.utils import Singleton
+
+logger = logging.getLogger(__name__)
+
+
+class LatencyMeasurement(NamedTuple):
+    """Individual latency measurement"""
+
+    model_id: UUID
+    latency_ms: float
+    timestamp: datetime
+
+
+class MetricsCollector(metaclass=Singleton):
+    """Thread-safe in-memory metrics collector for model latency data"""
+
+    def __init__(self, max_age_seconds: int = 60):
+        self._max_age_seconds = max_age_seconds
+        self._measurements: deque[LatencyMeasurement] = deque()
```

Comment:
The inference worker runs in a separate process, meaning a different Python interpreter instance with isolated memory, and will have its own Singleton instance of `MetricsCollector`. As a result, the measurements deque will be different from the one in the main process where the FastAPI server is started. Using an SQLite database would be the easiest way to persist the measurements. Are there requirements to keep it in memory?